### ISSUE TITLE
[BUG] [MSInstallation isEqualToMSInstallation:] is passing nil to NSCalendar method

### ISSUE THREAD

**Issue by GyroJoe** [Azure/azure-notificationhubs-ios#112]
**Describe the bug**
[MSInstallation isEqualToMSInstallation:] is passing nil to NSCalendar method. This is generating the following log spew from Apple, which indicates this will be an exception in the future:
*** -[NSCalendar isDate:equalToDate:toUnitGranularity:]: date1 cannot be nil
Future exception.
A few of these errors are going to be reported with this complaint, then further violations will simply be ignored.

***Exception or Stack Trace***
```
CoreFoundation 0x00000001a68b4168 935533F2-35EE-314E-A760-E74521F68435 + 1278312
WindowsAzureMessaging 0x0000000102ca3724 -[MSInstallation isEqualToMSInstallation:] + 588
WindowsAzureMessaging 0x0000000102ca3d30 -[MSInstallation isEqual:] + 204
WindowsAzureMessaging 0x0000000102c9ba40 -[MSDebounceInstallationManager execute] + 120
```

This codepath should nil check before calling the NSCalendar API.

**To Reproduce**
1. Start MSNotificationHub
2. Set the user ID

***Code Snippet***
Seems to ultimately be triggered by setting the user ID:
```
MSNotificationHub.setUserId("\(userId.lowercased())")
```

**Expected behavior**

**Setup (please complete the following information):**
 - OS: iOS 14.3
 - IDE : Xcode 12.3
 - WindowsAzureMessaging 3.1.1 (SwiftPM)

**Additional context**
```
(lldb) po self.expirationTime
 nil
(lldb) po installation.expirationTime
2021-04-06 02:55:33 +0000
```

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

**Comment by mpodwysocki** [comment]
@GyroJoe Yes, this seems to be an issue with Xcode, luckily it does not cause a crash but yet seems now.  Will investigate

**Comment by mpodwysocki** [comment]
Closed by #108
